crates/interpreter/src/module.rs

```diff
@@ -65,8 +69,10 @@ impl ImportDesc {
 impl<'m> Module<'m> {
     /// Validates a WASM module in binary format.
     pub fn new(binary: &'m [u8]) -> Result<Self, Error> {
-        validate(binary)?;
-        Ok(unsafe { Self::new_unchecked(binary) })
+        let side_tables = validate(binary)?;
+        let mut module = unsafe { Self::new_unchecked(binary) };
```

Comment:
Ideally `new_unchecked` should take the side tables, but this will break the scheduler. But this issue is temporary until we persist the side tables.